### PR TITLE
Bubblegum: use @solana/spl-account-compression

### PR DESCRIPTION
**Pull request by ngundotra**
Replace hard coded tree-size by calculating account size with `@solana/spl-account-compression`

### CODE REVIEW

File: bubblegum/js/tests/main.test.ts [hunk 1]
```diff
@@ -36,14 +42,17 @@ describe("Bubblegum tests", () => {
         const merkleTreeKeypair = Keypair.generate();
         const merkleTree = merkleTreeKeypair.publicKey;
 
+        const maxDepth = 14;
+        const maxBufferSize = 64;
+
         // Hard code space until @solana/spl-account-compression released
```
Comment (danenbm) [inline bubblegum/js/tests/main.test.ts:48]: Nit: outdated comment

**Review by danenbm — CHANGES_REQUESTED** [review]

File: bubblegum/js/tests/main.test.ts (hunk 1 above)
Comment (ngundotra) [reply bubblegum/js/tests/main.test.ts]: Fixed

**Review by danenbm — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -39,5 +39,8 @@
     "eslint-plugin-prettier": "^4.0.0",
     "lerna": "^4.0.0",
     "prettier": "^2.4.1"
+  },
+  "dependencies": {
+    "@solana/spl-account-compression": "^0.1.0"
```
Comment (stranzhay) [inline package.json:44]: should this be 0.1.1 or 0.1.2

Comment (ngundotra) [reply package.json:44]: Not yet published

File: bubblegum/js/tests/main.test.ts [hunk 3]
```diff
@@ -21,8 +27,8 @@ function keypairFromSeed(seed: string) {
     return Keypair.fromSeed(expandedSeed.slice(0, 32));
 }
 
-const LOG_WRAPPER_ID = new PublicKey("noopb9bkMVfRPU8AsbpTUg8AQkHtKwMYZiFUjNRtMmV");
-const COMPRESSION_PROGRAM_ID = new PublicKey("cmtDvXumGCrqC1Age74AVPhSRVXJMd8PJS91L8KbNCK");
+// const LOG_WRAPPER_ID = new PublicKey("noopb9bkMVfRPU8AsbpTUg8AQkHtKwMYZiFUjNRtMmV");
+// const COMPRESSION_PROGRAM_ID = new PublicKey("cmtDvXumGCrqC1Age74AVPhSRVXJMd8PJS91L8KbNCK");
```
Comment (stranzhay) [inline bubblegum/js/tests/main.test.ts:31]: can these be removed if not needed any more

File: package.json (hunk 2 above)
Comment (ngundotra) [reply package.json:44]: I'm assuming that I'll need to push some changes to the JS sdk while writing Bubblegum tests, so I just set to ^0.1.0

Comment (stranzhay) [reply package.json:44]: ok sounds good

File: bubblegum/js/tests/main.test.ts (hunk 3 above)
Comment (ngundotra) [reply bubblegum/js/tests/main.test.ts]: Removed

**Review by danenbm — APPROVED** [review]

**Review by stranzhay — APPROVED** [review]

**Pull request merged**